kodilla-stream/src/main/java/com/kodilla/stream/StreamMain.java

```diff
@@ -52,6 +68,64 @@ public static void main(String[] args) {
             }
             return result.toString();
         };
-        poemBeautifier.beautify(baseText, lambda);
+        poemBeautifier.beautify(baseText, lambda);*/
+
+        /*  modul 7.3 - kurs
+        People.getList().stream()
+
+                .map(String::toUpperCase)
+                .filter(s -> s.length() > 11)
+                .map(s -> s.substring(0, s.indexOf(' ') + 2) + ".")
+                .filter(s -> s.startsWith("M"))
+                .forEach(System.out::println);
+        //-------------------------------------------------------------
+        BookDirectory theBookDirectory = new BookDirectory();
+        theBookDirectory.getList().stream()
+                .filter(book -> book.getYearOfPublication() > 2005)
+                .forEach(System.out::println);
+        //-------------------------------------------------------------
+        BookDirectory theBookDirectory = new BookDirectory();
+        List<Book> theResultListOfBooks = theBookDirectory.getList().stream()
+                .filter(book -> book.getYearOfPublication() > 2005)
+                .collect(Collectors.toList());
+
+        System.out.println("# elements: " + theResultListOfBooks.size());
+        theResultListOfBooks
+                .forEach(System.out::println);
+        //-------------------------------------------------------------
+
+        BookDirectory theBookDirectory = new BookDirectory();
+
+        Map<String, Book> theResultMapOfBooks = theBookDirectory.getList().stream()
+                .filter(book -> book.getYearOfPublication() > 2005)
+                .collect(Collectors.toMap(Book::getSignature, book -> book));
+
+        System.out.println("# elements: " + theResultMapOfBooks.size());
+        theResultMapOfBooks.entrySet().stream()
+                .map(entry -> entry.getKey() + ": " + entry.getValue())
+                .forEach(System.out::println);
+        //-------------------------------------------------------------
+
+        BookDirectory theBookDirectory = new BookDirectory();
+        String theResultStringOfBooks = theBookDirectory.getList().stream()
+                .filter(book -> book.getYearOfPublication() > 2005)
+                .map(Book::toString)
+                .collect(Collectors.joining(",\n","<<\n","\n>>"));
+
+        System.out.println(theResultStringOfBooks);
+        */
+
+        // Modul 7.3 Zadanie: funkcyjny spacer po liście użytkowników forum
+        Forum forum = new Forum();
+        Map<Integer, ForumUser> userMap = forum.getUserList().stream()
+                .filter(forumUser -> forumUser.getSex() == 'M')
+                .filter(forumUser -> forumUser.getBirthDate().isBefore(LocalDate.of(LocalDate.now().getYear()-20,
+                        LocalDate.now().getMonth(), LocalDate.now().getDayOfMonth())))
+                .filter(forumUser -> forumUser.getPostCounter() > 0)
+                .collect(Collectors.toMap(ForumUser::getId, forumUser -> forumUser));
```

Comment:
jak potrzebujesz żeby funkcja zwracała po prostu argument wejściowy (tutaj `forumUser -> forumUser`) to zamiast tego możesz użyć specjalnie przeznaczonej do tego metody `Function.identity()`, czyli będzie `.toMap(ForumUser::getId,Function.identity())`